src/scripts.js

```diff
@@ -150,10 +181,32 @@ function toggleFavorites(event) {
   })
 }
 
+function displayRedHearts(list) {
+  const heartButtons = document.querySelectorAll('.fa-heart');
+  list.forEach(recipe => {
+    user.favorites.includes(recipe) ? 
+      heartButtons.forEach(button => {
+        button.parentNode.parentNode.parentNode.id === 'id' + recipe.id ? 
+          button.classList.add('red') : null;
+      }) : null;
+  });
+}
+
+function displayYellowBookmarks(list) {
+  const saveButtons = document.querySelectorAll('.fa-bookmark');
+  list.forEach(recipe => {
+    user.cookbook.includes(recipe) ? 
+      saveButtons.forEach(button => {
+        button.parentNode.parentNode.parentNode.id === 'id' + recipe.id ? 
+          button.classList.add('yellow') : null;
+      }) : null;
+  });
+}
+
 function toggleCookbook(event) {
   const saveButtons = document.querySelectorAll('.fa-bookmark');
   const recipeId = event.target.parentNode.parentNode.parentNode.id;
-  const thisRecipe = recipeRepo.currentRecipes.find(recipe => "id" + recipe.id === recipeId);
+  const thisRecipe = recipeRepo.recipes.find(recipe => "id" + recipe.id === recipeId);
```

Comment:
Find the recipe in the list of all recipes, not the current recipes. We're just looking for the id here